tls/handshake_client.go

```diff
@@ -493,7 +493,7 @@ func (c *Conn) clientHandshake() error {
 	if err != nil {
 		return err
 	}
-
+	c.buffering = true
```

Comment:
Do we need to `defer` a flush?

Reply:
That would probably be a good idea -- a panic with buffering = true would be pretty catastrophic. It's not in the original patch, so I'll drop it in the other branch.

Reply:
@dadrian -
 https://github.com/zmap/zcrypto/compare/feature/bufferHandshakes...feature/gatedBufferHandshakes#diff-00b59709a41dd3e0bfe7ade1ee596779R498